opencog/benchmark/AtomSpaceBenchmark.cc

```diff
@@ -987,7 +994,7 @@ timepair_t AtomSpaceBenchmark::bm_getHandleSet()
     case BENCH_AS: {
         HandleSeq results;
         clock_t t_begin = clock();
-        asp->getHandlesByType(back_inserter(results), t, true);
+        asp->getHandlesByType(results, t, true);
```

Comment:
Here's the new caller, just pass the results in as the first parameter since we're not using the iterators in a loop.

This case is similar to the one that will speed up Python as that's what it is currently doing for many calls, getting a big list of Handles and then copying them. So being able to not have to copy the whole vector from C++ into Python world will be a big win for the yield iterators in Python as we'll be able to iterate over the underlying Handle vector directly in Cython code. The Cython code will be able to pass it's vector into the AtomSpace if we use this approach eliminating a copy. There may be other ways to improve it too, but this way is an easy change compared to the other ways.